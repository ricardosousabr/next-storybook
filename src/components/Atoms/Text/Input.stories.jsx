import Text from "./index.jsx";

export default {
  title: "Components/Text",
  Component: Text,
  parameters: {
    backgrounds: {
      values: [
        { name: "dark", value: "#000000" },
        { name: "white", value: "#ffffff" },
        { name: "purple", value: "#b220d7" },
      ],
    },
  },
};

export const Primary = (args) => <Text {...args} />;
Primary.args = {
  children: "",
  color: "red",
  fontSize: "12px",
  as: "p",
};
