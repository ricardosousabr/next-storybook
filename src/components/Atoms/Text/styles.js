import { styled } from "styled-components";

export const TextStyle = styled.p`
  color: ${(props) => props.color};
  font-size: ${(props) => props.fontSize};
`