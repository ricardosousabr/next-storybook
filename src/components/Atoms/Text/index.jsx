import { TextStyle } from "./styles.js";
import PropTypes from "prop-types";

export default function Text({children, color, fontSize, as}) {
  return <TextStyle as={as} color={color} fontSize={fontSize}>{children}</TextStyle>
}

Text.prototype = {
  children: PropTypes.string,
  color: PropTypes.string,
  fontSize: PropTypes.string,
  as: PropTypes.oneOf(["p", "h1", "h2", "h3", "h4", "h5", "h6"])
}
