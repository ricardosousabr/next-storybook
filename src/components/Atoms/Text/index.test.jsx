import renderer from "react-test-renderer";
import Text from "./index";

test("render text", () => {
  const tree = renderer.create(<Text color={"aqua"} fontSize={"12px"}>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Ipsam vel quas, iusto nulla aut temporibus optio sed! Asperiores deleniti at sed illo neque, eveniet est eius, itaque sint, vitae molestias.</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});
