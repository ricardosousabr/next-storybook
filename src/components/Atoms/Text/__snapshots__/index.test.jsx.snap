// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render text 1`] = `
<p
  className="styles__TextStyle-sc-15myn20-0 hdLYJM"
  color="aqua"
  fontSize="12px"
>
  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Ipsam vel quas, iusto nulla aut temporibus optio sed! Asperiores deleniti at sed illo neque, eveniet est eius, itaque sint, vitae molestias.
</p>
`;
